### PR TITLE
Add SEP for metadata handling

### PR DESCRIPTION
**Pull request by dstansby**
This has spun out of https://github.com/sunpy/sunpy/issues/4410. Very much a draft, and discussion and suggestions are welcome. The general approach was suggested by @ayshih in https://github.com/sunpy/sunpy/issues/4410#issuecomment-689935291

### CODE REVIEW

**Comment by ayshih** [comment]
Instead of allowing source-specific loaders to be completely silent when fixing metadata, what if we require *any* metadata fix to produce at least a corresponding debug-level logger message?  At some level, this would force source-specific loaders to be "accountable" for metadata issues since users can easily turn on the debug messages.

File: SEP-0010.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+## Proposal
+This SEP proposes the following process for handling metadata:
+
+1. Generic data loaders
+
+The sunpy core package contains base types for storing data, e.g. GenericMap,
+GenericTimeseries. Trying to load data that is not standards compliant with
+these types should raise an error as soon as possible in the data loading
+process. This allows users and developers to be easily notified of non-compliant
+metadata, and follows fail fast approach.
```
Comment (rhewett) [inline SEP-0010.md:30]: I wonder if the default mode should be error, but if an there could/should be a keyword that either changes exception to warning or allows non-compliant metadata to be loaded.

I can easily conceive of situations where there is older data to be loaded that may not have come through a standard instrument pipeline, or that there is not an instrument specific class for.  Under this framework, if that data has noncompliant metadata, there is no mechanism to load it or to force it to be loaded.

An alternative would be to provide a `NoncompliantMap` interface.  Either way, the user would be forced to explicitly acknowledge that their data is not compliant and SunPy would not be obligated to do anything "correct" with it.

Comment (dstansby) [reply SEP-0010.md]: The current mechanism for manually fixing metadata is documented here: https://docs.sunpy.org/en/stable/code_ref/map.html#fixing-map-metadata Essentially, load the data with a fits reader, fix the metadata, and then give it to `Map`. I think this is a reasonable tradeoff, and if fixes are common enough a new map class can be created.

**Comment by dstansby** [comment]
This is redundant in light of https://github.com/sunpy/sunpy/pull/5501, which takes a different approach (which was discussed and agreed upon at the 2021 developer meeting)

**Pull request closed without merging**